lib/main.dart

```diff
@@ -92,18 +99,50 @@ List<AppLauncherPanelButton> testLaunchers = [
       app: Settings(), icon: 'lib/images/icons/v2/compiled/settings.png'),
 ];
 
-void main() {
+void main() async {
+  //init hive
+  WidgetsFlutterBinding.ensureInitialized();
+  Directory dir = await getApplicationDocumentsDirectory();
+  Hive.init(dir.path);
+  await Hive.openBox<String>("settings");
+
   /// To keep app in Portrait Mode
   //SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeRight]);
-  runApp(MyApp());
+  runApp(Pangolin());
+}
+
+class Pangolin extends StatefulWidget {
+  @override
+  _PangolinState createState() => _PangolinState();
+
+  static void setLocale(BuildContext context, Locale locale) {
+    _PangolinState state = context.findAncestorStateOfType<_PangolinState>();
+    state.setLocale(locale);
+  }
+
+  static Box<String> settingsBox;
 }
 
-class MyApp extends StatelessWidget {
+class _PangolinState extends State<Pangolin> {
+  Locale _locale;
+
+  @override
+  void initState() {
+    Pangolin.settingsBox = Hive.box("settings");
+    _locale = Locale(Pangolin.settingsBox.get("language"));
```

Comment:
On a fresh clone not checking if Pangolin.settingsBox.get("language") exists returns in `flutter: 'dart:ui/window.dart': Failed assertion: line 285: '<optimized out>': is not true.` error, if you set it to null, run it and set it in quick settings with a button, and then restore this line, the error is no more because the "language" property already exists in the box, which is probably why you're not getting this error.

Suggested fix could look like:
```suggestion
    if (Pangolin.settingsBox.get("language") != null)
      _locale = Locale(Pangolin.settingsBox.get("language"));
    else 
      _locale = null;
```